src/events.js

```diff
@@ -70,6 +81,51 @@ module.exports = function(ctx) {
     currentMode.mouseout(event);
   };
 
+  events.touchstart = function(event) {
+    // Prevent emulated mouse events because we will fully handle the touch here.
+    // This does not stop the touch events from propogating to mapbox though.
+    event.originalEvent.preventDefault();
+    if (!ctx.options.touchEnabled) {
+      return;
+    }
+
+    touchStartInfo = {
+      time: new Date().getTime(),
+      point: event.point
+    };
+    var target = featuresAt.touch(event, null, ctx)[0];
+    event.featureTarget = target;
+    currentMode.touchstart(event);
```

Comment:
@ChrisBellew what are your thoughts on mapping touch events and map events to the same event handlers? Is there a time when a `touchstart` event is not the same as a `mousedown` event?

One place I could see this as being different is in multi-select. A `tap` event and a `click` event might want to work differently to enable touch-based multi-select. That said, I think this can still be down my mapping the events to the same handler. Rather than passing the raw event to the handler, we'd need to pass our the interpreted event.